app/modules/net_info.c

```diff
@@ -0,0 +1,176 @@
+	       ipaddrstr,
+	       pingresp->seqno,
+	       pingresp->ttl,
+	       pingresp->resp_time);
+    }
+}
+
+
+
+static void ping_by_hostname(const char *name, ip_addr_t *ipaddr, void *arg) {
+    struct ping_option *ping_opt = (struct ping_option *)c_zalloc(sizeof(struct ping_option));
+
+    if (ipaddr == NULL) {
+      c_printf("SEVERE problem resolving hostname - network and DNS accessible?\n");
+        return;
+    }
+    if (ipaddr->addr == IPADDR_NONE) {
+      c_printf("problem resolving hostname - maybe nonexistent host?\n");
+	return;
+    }
+    
+    ping_opt->count = ping_host_count;
+    ping_opt->ip = ipaddr->addr;
+    ping_opt->coarse_time = 0;
+    ping_opt->recv_function = &ping_received;
+    
+    ping_start(ping_opt);
+}
+
+
+/**
+  * test.ping()
+  * Description:
+  * 	Send ICMP ping request to address, optionally call callback when response received
+  *
+  * Syntax:
+  * 	wifi.sta.getconfig(ssid, password) --Set STATION configuration, Auto-connect by default, Connects to any BSSID
+  *     test.ping(address)              -- send 4 ping requests to target address
+  *     test.ping(address, n)           -- send n ping requests to target address
+  *     test.ping(address, n, callback) -- send n ping requests to target address
+  * Parameters:
+  * 	address: string 
+  * 	n: number of requests to send
+  * 	callback: 
+  * Returns:
+  * 	Nothing.
+  *
+  * Example:
+  *     test.ping("192.168.0.1")               -- send 4 pings to 192.168.0.1
+  *     test.ping("192.168.0.1", 10)           -- send 10 pings to 192.168.0.1
+  *     test.ping("192.168.0.1", 10, got_ping) -- send 10 pings to 192.168.0.1, call got_ping() with the
+  *                                           --     ping results
+  */
+
+static int net_info_ping(lua_State *L)
+{
+    const char *ping_target;
+    unsigned count = 4;
+
+    // retrieve address arg (mandatory)
+    if (lua_isstring(L, 1)) {
+	ping_target = luaL_checkstring(L, 1);
+    } else {
+	return luaL_error(L, "no address specified");
+    }
+
+    // retrieve count arg (optional)
+    if (lua_isnumber(L, 2)) {
+	count = luaL_checkinteger(L, 2);
+    }
+   
+    // retrieve callback arg (optional)
+    if (ping_callback_ref != LUA_NOREF) 
+      luaL_unref(L, LUA_REGISTRYINDEX, ping_callback_ref);
+    ping_callback_ref = LUA_NOREF;
+       
+    if (lua_type(L, 3) == LUA_TFUNCTION || lua_type(L, 3) == LUA_TLIGHTFUNCTION)
+      ping_callback_ref = luaL_ref(L, LUA_REGISTRYINDEX);
+    
+    gL = L;   // global L
+
+
+    // attempt to parse ping target as IP
+    uint32 ip = ipaddr_addr(ping_target);
+
+    if (ip != IPADDR_NONE) {
+	struct ping_option *ping_opt = (struct ping_option *)c_zalloc(sizeof(struct ping_option));
+
+	ping_opt->count = count;
+	ping_opt->ip = ip;
+	ping_opt->coarse_time = 0;
+	ping_opt->recv_function = &ping_received;
+	
+	ping_start(ping_opt);
+	
+    } else {
+	ping_host_count = count;
+
+	struct espconn *ping_dns_lookup;
+	espconn_create(ping_dns_lookup);
```

Comment:
This doesn't make sense -- ping_dns_lookup is not initialized. Also the return code of espconn_create is not checked.